src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -301,19 +307,17 @@ class PrisonerIepLevelReviewService(
               withDetails = true,
               useClientCredentials = true
             ).iepDetails
-          val iepLevelBeforeTransfer =
+          val levelCodeBeforeTransfer =
             iepHistory.sortedBy(IepDetail::iepTime).lastOrNull { it.agencyId != prisonerInfo.agencyId }?.iepCode
-              ?: defaultLevel // if no previous prison
-          iepLevelsForPrison.find { it.iepLevel == iepLevelBeforeTransfer }
-            ?: defaultLevel // if we don't match a level - go to the default.
+              ?: defaultLevelCode // if no previous prison
+          iepLevelService.findNearestHighestLevel(prisonerInfo.agencyId, levelCodeBeforeTransfer)
```

Comment:
We could pass `iepLevelsForPrison` into this function to avoid needing to duplicate the lookup?